app/views/api/v2/layouts/permissions.json.rabl

```diff
@@ -0,0 +1,8 @@
+object @resource
+
+if params.has_key?(:include_permissions)
+  # To avoid loading all the records into the cache
+  # authorized_for is used instead of authorizer.can?.
+  node(:can_edit) { |resource| authorized_for(:auth_object => resource, :authorizer => authorizer, :permission => "edit_#{controller_permission}") }
```

Comment:
Can we move this to methods on the models please? Otherwise, plugins have a hard time overriding this logic and we want core to be plugin friendly, don't we?

So this would only call `node(:can_edit) { |resource| resource.include?(Authorizable) && resource.can_edit? }`

Reply:
It will probably work, I'm more concerned about actions like `api/compute_resources/1/available_clusters` that aren't backed by a rails model.